src/Runner.Worker/Handlers/StepHost.cs

```diff
@@ -234,5 +281,28 @@ public async Task<string> DetermineNodeRuntimeVersion(IExecutionContext executio
                                                          cancellationToken: cancellationToken);
             }
         }
+
+        private string CheckPlatformForAlpineContainer(IExecutionContext executionContext, string preferredVersion)
+        {
+            string nodeExternal = preferredVersion;
+            if (!Constants.Runner.PlatformArchitecture.Equals(Constants.Architecture.X64))
+            {
+                var os = Constants.Runner.Platform.ToString();
+                var arch = Constants.Runner.PlatformArchitecture.ToString();
+                var msg = $"JavaScript Actions in Alpine containers are only supported on x64 Linux runners. Detected {os} {arch}";
+                throw new NotSupportedException(msg);
```

Comment:
should we remove this check if running in hooks is an interesting question. Can we add a todo? May make sense to be more permissive (we don't care if the runner is alpine if we are running in k8s)

Reply:
As a TODO: container hooks, for all we know, could be implemented such that the containers are running on a completely different machine. Should we even have _any_ restrictions?

Reply:
We can always loosen restrictions as we go forward, we can leave this for now.